Source/CrossingDetector/CrossingDetector.cpp

```diff
@@ -338,55 +406,41 @@ bool CrossingDetector::shouldTrigger(const float* rpCurr, int nSamples, int t0,
         return shouldTrigger(rpCurr, nSamples, t0, currThresh, true, false, currPastSpan, currFutureSpan)
         || shouldTrigger(rpCurr, nSamples, t0, currThresh, false, true, currPastSpan, currFutureSpan);
```

Comment:
Here and in the declaration of `shouldTrigger`, you can get rid of the parameters `rpCurr`, `nSamples`, `t0`, `currThresh`, `currPastSpan`, and `currFutureSpan` since the new implementation doesn't require any of these.

Reply:
Also not outdated